### DISCUSSION TITLE
600-odd lines of Homebrew related stuff in my .zprofile file, I think it's slowing my terminal startup down.

### DISCUSSION THREAD

**Discussion by Southpaw1496** [Homebrew/discussions#1855] [Everyday usage] [answered]
I've been having a problem where my terminal takes about six seconds to launch. While looking through my home folder for things that might be causing this, I found that the `.zprofile` file has 600 lines of basically this:
shell
```
echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> /Users/john/.zprofile
    eval $(/opt/homebrew/bin/brew shellenv)
eval $(/opt/homebrew/bin/brew shellenv)
eval $(/opt/homebrew/bin/brew shellenv)
eval $(/opt/homebrew/bin/brew shellenv)
eval $(/opt/homebrew/bin/brew shellenv)

```
I think I should probably get rid of it (and possibly re-do my entire Homebrew installation?) but I don't know if bad things will happen if I delete all that, so I thought I'd check here first.

**Comment by carlocab** [comment, marked as answer]
You can delete all of them. Just leave one
```
eval $(/opt/homebrew/bin/brew shellenv)
```

The problem seems to be that you have an
```
echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> /Users/redacted/.zprofile
```
in your `.zprofile`, which creates a new
```
eval $(/opt/homebrew/bin/brew shellenv)
```
line in your `.zprofile` every time your shell starts. You'll want to remove that line too, so that you don't have this filling up your `.zprofile` again.

As an aside, please [don't use screenshots](https://meta.stackoverflow.com/a/285557) when sharing text on your screen. They're often difficult to read and impossible to reuse.
